tentacle/src/transports/mod.rs

```diff
@@ -116,15 +118,44 @@ mod os {
         MemoryDialFuture, MemoryListenFuture, MemoryListener, MemorySocket, MemoryTransport,
     };
     use self::tcp::{TcpDialFuture, TcpListenFuture, TcpTransport};
+    #[cfg(feature = "tls")]
+    use self::tls::{TlsDialFuture, TlsListenFuture, TlsListener, TlsStream, TlsTransport};
     #[cfg(feature = "ws")]
     use self::ws::{WebsocketListener, WsDialFuture, WsListenFuture, WsStream, WsTransport};
+    #[cfg(feature = "tls")]
+    use tokio_rustls::rustls::{ClientConfig, ServerConfig};
+
+    #[derive(Clone, Default)]
+    #[cfg(feature = "tls")]
+    pub struct TlsConfig {
+        pub(crate) tls_server_config: Option<ServerConfig>,
+        pub(crate) tls_client_config: Option<ClientConfig>,
+        pub(crate) tls_bind: Option<SocketAddr>,
+    }
+
+    #[cfg(feature = "tls")]
+    impl TlsConfig {
+        pub fn new(
+            tls_server_config: Option<ServerConfig>,
+            tls_client_config: Option<ClientConfig>,
+            tls_bind: Option<SocketAddr>,
```

Comment:
Abstract leaked, here can only be multi-addr, move this structure under service/config.rs please

Reply:
tcp and ws are socket addr